src/mesh/Router.cpp

```diff
@@ -317,6 +318,7 @@ bool perhapsDecode(meshtastic_MeshPacket *p)
             memcpy(bytes, p->encrypted.bytes,
                    rawSize); // we have to copy into a scratch buffer, because these bytes are a union with the decoded protobuf
             crypto->decrypt(p->from, p->id, rawSize, bytes);
+            p->channel = chIndex; 
```

Comment:
in encrypted packets the channel index is set by the sending node which have to set to local channel index AFTER decoding (unwrapping)